addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -310,20 +310,27 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
                 last_orderline.set_discount(code.value);
             }
         }
+        async _parseElementsFromGS1(parsed_results) {
+            const productBarcode = parsed_results.find(element => element.type === 'product');
+            const lotBarcode = parsed_results.find(element => element.type === 'lot');
+            let options={}
+            const product = await this._getProductByBarcode(productBarcode);
+            return {product:product, lotBarcode:lotBarcode, options:options }
+        }
         /**
          * Add a product to the current order using the product identifier and lot number from parsed results.
          * This function retrieves the product identifier and lot number from the `parsed_results` parameter.
          * It then uses these values to retrieve the product and add it to the current order.
          */
         async _barcodeGS1Action(parsed_results) {
-            const productBarcode = parsed_results.find(element => element.type === 'product');
-            const lotBarcode = parsed_results.find(element => element.type === 'lot');
-            const product = await this._getProductByBarcode(productBarcode);
+            const GS1options = await this._parseElementsFromGS1(parsed_results)
+            const product = GS1options.product;
             if (!product) {
                 return;
             }
-            const options = await this._getAddProductOptions(product, lotBarcode);
-            await this.currentOrder.add_product(product, options);
+            const options = await this._getAddProductOptions(GS1options.product, GS1options.lotBarcode);
+            let productOptions = {...options,...GS1options.options}
+            await this.currentOrder.add_product(product, productOptions);
```

Comment:
```suggestion
            const option = await this._getAddProductOptions(product, lotBarcode);
            await this.currentOrder.add_product(product, options);
```

Reply:
Hello,
i did it as you see in the PR because in getAddProductOptions you can't access the GS1 parsed elements... 
Imagine there is a need to parse for example the discount gs1 rule  with an additional module .... With the code you provided you can't use the parsed discount value because you have not access inside the following function(and so on if somebody needs to parse other rule) 
` async _barcodeGS1Action(parsed_results) {`
so i imagined to extend the options with an additional options from the parseGS1 function.
The main reason of parseElements function returning options dict (default empty , so everybody can override and change options) is to make the developing more flessible

Reply:
Oh, you are overriding the options from `_getAddProductOptions`. I adapted my suggestions.

Reply:
@caburj hello, i changed the code!